src/InstrumentationEngine/MethodInfo.cpp

```diff
@@ -1124,28 +1125,28 @@ HRESULT MicrosoftInstrumentationEngine::CMethodInfo::InitializeName(_In_ mdToken
 
     if (!m_genericParameters.empty())
     {
-        tstringstream nameBuilder;
```

Comment:
I'm a little bit concerned with what the new performance characteristics of this change could be. Potentially hundres of thousands of method names could be initialized by the instrumentation engine, and doing string concatination like this may be slow.

Reply:
AFAIK it isn't any different than what stringstream would be doing internally, which is pretty much string appending + formatting with stream semantics (which is actually more work if you think about this).

If you're concerned about this, it might make sense to prereserve some space in the string so it doesn't usually have to reallocate the buffer. However, this can turn out to be a pessimisation especially in the case where most method names fit into a small string buffer (although I don't think this is likely).

Reply:
Sorry for the slow follow-up.

You are right. I double-checked  the disassembly for this change, and it shouldn't be any worse that string streams.

I might recommend removing tstringstream altogether, then, so that we don't accidentally use it in locations that this might cause a problem.

Reply:
I think tstringstream might be used in tests, so you could move it into the tests. We can figure out what to do with it as we update the tests to run on other platforms.